docs/AVS-Guide.md

```diff
@@ -0,0 +1,85 @@
+
+3. *Services Slash Only Objectively Attributable Behavior*: 
+    EigenLayer is built to support slashing as a result of an on-chain-checkable, objectively attributable action. An AVS SHOULD slash in EigenLayer only for such provable and attributable behavior. It is expected that operators will be very hesitant to opt-in to services that slash for other types of behavior, and other services may even choose to exclude operators who have opted-into serving one or more AVSs with such “subjective slashing conditions”, as these slashing conditions present a significant challenge for risk modeling, and may be perceived as more dangerous in general. Some examples of on-chain-checkable, objectively attributable behavior: 
+    - double-signing a block in Ethereum, but NOT inactivity leak; 
+    - proofs-of-custody in EigenDA, but NOT a node ceasing to serve data; 
+    - a node in a light-node-bridge AVS signing an invalid block from another chain.
+
+4. *Single Point-of-Interaction for Services and EigenLayer*: 
+    It is assumed that services have a single contract that coordinates the service’s communications sent to EigenLayer. This contract – referred to as the ServiceManager – informs EigenLayer of operator registration, updates, and deregistration, as well as signaling to EigenLayer when an operator should be slashed (frozen). An AVS has full control over how it splits the actual logic involved, but is expected to route all calls to EigenLayer through a single contract. While technically possible, an AVS SHOULD NOT use multiple contracts to interact with EigenLayer due to a couple of reasons. Firstly, multiple contracts will impose additional burden on stakers in EigenLayer to withdrawing stake. Secondly, operators will be faced with the complexity of selecting the correct contract in order to deregister from the AVS.
+
+## Integration with EigenLayer Contracts:
+In this section, we will explain various API interfaces that EigenLayer provides which are essential for AVSs to integrate with EigenLayer. 
+
+### *Opting into AVS*
+In order for any EigenLayer operator to be able to opt-in to a AVS, EigenLayer provides two interfaces: `optIntoSlashing(..)` and `recordFirstStakeUpdate(..)`. The sequential flow for opting into a AVS using these functions is as follows:
+1. The operator first opts into slashing by calling  `optIntoSlashing(..)` in `Slasher.sol`, where it has to specify the address of the AVS's ServiceManager contract in the argument. This step results in the operator giving permission to the AVS's ServiceManager contract to slash the operator via EigenLayer, if the operator is ever proven to have engaged in adversarial behavior while responding to the AVS's task. Successful call to  `optIntoSlashing(..)` emits the `OptedIntoSlashing(..)` event.
```

Comment:
typo: inconsistent pluralization -- "Successful call to  `optIntoSlashing(..)` emits" => "A successful call to  `optIntoSlashing(..)` emits"